### ISSUE TITLE
[TT-9364] jwt_claims_user_id in long data type returns incorrect value

### ISSUE THREAD

**Issue by mortymacs** [TykTechnologies/tyk#5134]
**Branch/Environment/Version**
- Branch/Version: v4.2.3
- Environment: On-prem

**Describe the bug**
`$tyk_context.jwt_claims_something` in the `long` datatype gives a different number, while when the data type is a string, it shows the correct number.

**Reproduction steps**
This is the app config:
```json
{
  "active": true,
  "api_id": "my-oidc",
  "enable_context_vars": true,
  "internal": true,
  "jwt_scope_claim_name": "my-roles,",
  "name": "my-oidc",
  "openid_options": {
    "providers": [
      {
        "client_ids": {
          "bXlvcmc=": "full-access"
        },
        "issuer": "http://localhost/auth/realms/services"
      }
    ]
  },
  "org_id": "default",
  "proxy": {
    "listen_path": "/",
    "strip_listen_path": true,
    "target_url": "https://httpbin"
  },
  "use_openid": true,
  "version_data": {
    "default_version": "v1",
    "not_versioned": false,
    "versions": {
      "v1": {
        "global_headers": {
          "X-User-Id": "$tyk_context.jwt_claims_user_id"
        },
        "name": "v1"
      }
    }
  }
}
```

**Actual behavior**
When I call the Keycloak API to get the user info, it returns the `user_id` claim with the correct number: `123452342`. But when it goes through Tyk, the `X-User-Id` header will contain `123452350`, for example.

**Expected behavior**
Tyk should set the actual `user_id` in the header which is `123452342`.

**Screenshots/Video**

**Logs (debug mode or log file):**
Nothing special.

**Configuration (tyk config file):**
Nothing special than the default config.

**Additional context**
The `user_id` contains a long number. When we change the data type on Keycloak to string, it will work fine. We're still determining what's happening there. Also, it's important to know that if the number is in a small range (e.g., 1500), it works fine, but when the number is big, it will give a different output.
It's still unknown why the `user_id` number in long will differ from when it's in a string or small number.

**Comment by mortymacs** [comment]
Hi,

Do you have an update on this issue?

Cheers,
Mort

**Comment by Bouwdie** [comment]
Just to double check you expect that the value changes for large numbers due to to some conversion logic in the Tyk codebase? :octopus:

**Comment by mortymacs** [comment]
> Just to double check you expect that the value changes for large numbers due to to some conversion logic in the Tyk codebase? 🐙

No, I was hoping to receive the exact same value as the one we obtained from Keycloak Claims.
I'm not 100% sure it's related to the data conversion, but the result shows it seems it has some sort of data conversion internally.

**Comment by mortymacs** [comment]
Hi @andyo-tyk 

No problem at all.
Thank you so much!

Cheers,
Mort

**Comment by andyo-tyk** [comment]
This will be fixed in Tyk 5.0.4 and 5.1.1.

**Comment by mortymacs** [comment]
> This will be fixed in Tyk 5.0.4 and 5.1.1.

Great! Thanks a million!
